src/config.ts

```diff
@@ -0,0 +1,73 @@
+import type {MergeableRecord} from './util/objects'
+import {deepMergeObject} from './util/objects'
+import path from 'path'
+import fs from 'fs'
+import {log} from './util/log'
+import {getParentDirectory} from './util/files'
+import Joi from 'joi'
+
+export interface FlowrConfigOptions extends MergeableRecord {
+	ignoreSourceCalls: boolean
+}
+
+export const defaultConfigOptions: FlowrConfigOptions = {
+	ignoreSourceCalls: false
+}
+export const defaultConfigFile = 'flowr.json'
+
+const schema = Joi.object({
+	ignoreSourceCalls: Joi.string().optional()
+})
+
+let configWorkingDirectory = process.cwd()
+let configFile = defaultConfigFile
+let currentConfig: FlowrConfigOptions | undefined
+
+export function setConfigFile(workingDirectory = process.cwd(), file = defaultConfigFile) {
+	configWorkingDirectory = workingDirectory
+	configFile = file
+
+	// reset the config so it's reloaded next time
+	currentConfig = undefined
+}
+
+export function setConfig(config: FlowrConfigOptions) {
+	currentConfig = config
+}
+
+export function getConfig(): FlowrConfigOptions {
```

Comment:
Sometime... we could rename this to flowrs global configuration :P

I understand why you made it lazy, but then, theoretically, as a user - i would assume that flowr would use the config file as it was when it started (e.g. when i am currently editing it to find out about its behavior) and not randomly whenever it is needed somewhere, so maybe we want to force the lazy eval in the main flowr scripts (especially to make it future proof with more and more configuration options being added :D)